12.LOCO_BULK_MLP/0.data_splits/scripts/1.data_splits_model_multiclass.py

```diff
@@ -0,0 +1,335 @@
+
+# ### Being a binary model this notebook will be limited to predicting one class 1 or 0, yes or no.
+# ### Here I will be predicting if a cell received a treatment or not
+
+# In[1]:
+
+
+import argparse
+import json
+import pathlib
+import sys
+
+import numpy as np
+import optuna
+import pandas as pd
+import pyarrow.parquet as pq
+import toml
+import torch
+from sklearn import preprocessing
+
+MLP_parent_path = pathlib.Path("../../../utils/")
+sys.path.append(str(MLP_parent_path.resolve()))
+MLP_path = pathlib.Path("../../../utils/MLP_utils").resolve()
+
+from MLP_utils.parameters import Parameters
+from MLP_utils.utils import (
+    Dataset_formatter,
+    data_split,
+    extract_best_trial_params,
+    objective_model_optimizer,
+    parameter_set,
+    plot_metric_vs_epoch,
+    results_output,
+    test_optimized_model,
+    train_optimized_model,
+    un_nest,
+)
+from sklearn.model_selection import train_test_split
+
+from utils import df_stats
+
+# In[2]:
+
+
+# set up the parser
+parser = argparse.ArgumentParser(description="Run hyperparameter optimization")
+parser.add_argument(
+    "--cell_type",
+    type=str,
+    default="all",
+    help="Cell type to run hyperparameter optimization for",
+)
+parser.add_argument(
+    "--model_name",
+    type=str,
+    default="all",
+    help="Model name to run hyperparameter optimization for",
+)
+
+# parse arguments
+args = parser.parse_args()
+
+CELL_TYPE = args.cell_type
+MODEL_NAME = args.model_name
+
+
+# In[3]:
+
+
+ml_configs_file = pathlib.Path(MLP_path / "multi_class_config.toml").resolve(
+    strict=True
+)
+ml_configs = toml.load(ml_configs_file)
+params = Parameters()
+mlp_params = parameter_set(params, ml_configs)
+
+# overwrite params via command line arguments from papermill
+mlp_params.CELL_TYPE = CELL_TYPE
+mlp_params.MODEL_NAME = MODEL_NAME
+MODEL_TYPE = mlp_params.MODEL_TYPE
+HYPERPARAMETER_BATCH_SIZE = mlp_params.HYPERPARAMETER_BATCH_SIZE
+
+
+# In[4]:
+
+
+# Import Data
+# set data file path under pathlib path for multi-system use
+
+file_path = pathlib.Path(
+    f"../../../data/{mlp_params.CELL_TYPE}_preprocessed_sc_norm_aggregated.parquet"
+).resolve(strict=True)
+
+df1 = pd.read_parquet(file_path)
+
+
+# In[5]:
+
+
+# get paths for toml files
+ground_truth_file_path = pathlib.Path(MLP_path / "ground_truth.toml").resolve(
+    strict=True
+)
+treatment_splits_file_path = pathlib.Path(MLP_path / "splits.toml").resolve(strict=True)
+# read toml files
+ground_truth = toml.load(ground_truth_file_path)
+treatment_splits = toml.load(treatment_splits_file_path)
+
+
+# In[6]:
+
+
+# get information from toml files
+apoptosis_groups_list = ground_truth["Apoptosis"]["apoptosis_groups_list"]
+pyroptosis_groups_list = ground_truth["Pyroptosis"]["pyroptosis_groups_list"]
+healthy_groups_list = ground_truth["Healthy"]["healthy_groups_list"]
+test_split_100 = treatment_splits["splits"]["data_splits_100"]
+test_split_75 = treatment_splits["splits"]["data_splits_75"]
```

Comment:
I am unfamiliar with these splits; what do the 100 and 75 references?

Reply:
100% test vs 75% test